### PR TITLE
Add instrumentation sample

### PR DESCRIPTION
**Pull request by bergundy**

### CODE REVIEW

**Review by sw-yx — APPROVED** [review]

File: instrumentation/src/worker.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  // @@@SNIPSTART typescript-worker-full-logging-setup
+  // Create a worker that will use the Core instance installed above
+  const worker = await Worker.create({
+    workflowsPath: require.resolve('./workflows'),
+    activities,
+    taskQueue: 'instrumentation',
+    // Install interceptors
+    interceptors: {
+      activityInbound: [(ctx) => new ActivityInboundLogInterceptor(ctx, activityLogger)],
+      workflowModules: [require.resolve('./workflows/interceptors')],
```
Comment (lorensr) [inline instrumentation/src/worker.ts:84]: Should we support custom bundling?

File: instrumentation/src/workflows/interceptors.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import {
+  WorkflowInterceptorsFactory,
+  WorkflowInboundCallsInterceptor,
+  WorkflowExecuteInput,
+  Next,
+} from '@temporalio/workflow';
+import { logger } from './logger';
+
+/** Logs Workflow executions and their duration */
+export class WorkflowInboundLogInterceptor implements WorkflowInboundCallsInterceptor {
```
Comment (lorensr) [inline instrumentation/src/workflows/interceptors.ts:11]: Does this need to be exported?

**Review by lorensr — COMMENTED** [review]
Cool stuff! Can add to root README, and maybe add `<details>` to bottom of `instrumentation/README` with [example worker output](https://gist.github.com/lorensr/dd26d49b56b97dfe4320d1e503684d04)

File: instrumentation/src/workflows/interceptors.ts (hunk 2 above)
Comment (bergundy) [reply instrumentation/src/workflows/interceptors.ts:11]: Removed the export so people don't get confused.

**Comment by bergundy** [comment]
> Cool stuff! Can add to root README, and maybe add `<details>` to bottom of `instrumentation/README` with [example worker output](https://gist.github.com/lorensr/dd26d49b56b97dfe4320d1e503684d04)

Good idea, I was already going to add to the README, also added the output.

File: instrumentation/src/worker.ts (hunk 1 above)
Comment (bergundy) [reply instrumentation/src/worker.ts:84]: Maybe we should but it'd make the sample a bit less concise.
We should perhaps consider having all-in-one dev + prod ready sample.

**Comment by bergundy** [comment]
Oops accidentally pushed the otel sample fix here

**Pull request merged**